cache/compression/compression.go

```diff
@@ -0,0 +1,105 @@
+package compression
+
+import (
+	"errors"
+	"io"
+	"io/fs"
+	"os"
+
+	"github.com/bitrise-io/go-utils/v2/command"
+	"github.com/bitrise-io/go-utils/v2/env"
+	"github.com/bitrise-io/go-utils/v2/log"
+)
+
+// Compress creates a compressed archive from the provided files and folders using absolute paths.
+func Compress(archivePath string, includePaths []string, logger log.Logger, envRepo env.Repository) error {
+	cmdFactory := command.NewFactory(envRepo)
+
+	tarArgs := []string{
+		"--use-compress-program",
+		"zstd --threads=0", // Use CPU count threads
+		"-P",               // Same as --absolute-paths in BSD tar, --absolute-names in GNU tar
+		"-cf",
+		archivePath,
+		"--directory",
```

Comment:
Would it be possible to add a comment to every line explaining what it does (and also the same for the extraction part)? I have to admit I have not really used tar before and had to google all of them. It feels like future me would be grateful for it. (Because I will already forget the them once I submit my review.) 😄 

We already did a similar thing for our zipping utilities:
https://github.com/bitrise-io/go-utils/blob/0d0978ade8bffeddc555ed12856a1a30c2a2dbc8/ziputil/ziputil.go#L68

Reply:
Fixed